lib/protocol/encryption.go

```diff
@@ -590,3 +595,27 @@ func isEncryptedParentFromComponents(pathComponents []string) bool {
 	}
 	return true
 }
+
+type folderKeyRegistry struct {
+	keys map[string]*[keySize]byte // folder ID -> key
+	mut  sync.RWMutex
+}
+
+func newFolderKeyRegistry() *folderKeyRegistry {
+	return &folderKeyRegistry{
+		keys: make(map[string]*[keySize]byte),
+	}
+}
+
```

Comment:
Don't really need a constructor for this one, nil map is fine until reset() happens

Reply:
I don't really want a nil-check in `folderKeyRegistry.get` though.

Reply:
Why would you need one?

Reply:
I'd expect `key, ok := r.keys[folder]` to panic. And without checking I am not 100% that `model.AddConnection` is guaranteed to initialize the passwords before any message comes in from a remote. Expressing that thought makes it clear that I should keep initializing passwords when creating the connection. Likely it's all irrelevant practically, as the early messages aren't encrypted, but still.

Reply:
It doesn't panic though, a nil map is equivalent to a non-nil but empty map when reading.

Reply:
Cool, wasn't aware of that. I didn't really internalize that aspect of go (treating nil as a valid state), I treat nil as "do not touch".
Any ways, it does now get initialized at connection construction time with a valid map of passwords.